### PR TITLE
Feat/registration

### PR DESCRIPTION
**Pull request by Zmicer-Yafremau**
#### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Test Case
- [ ] Documentation update
- [ ] Other

#### 🔗 Related trello card link

[_Describe the source of requirement, like related issue link._](https://trello.com/c/xC7p5720/8-social-network-registration)

#### 💡 Background and solution

Finish registration

-create user on server
-get token
-change header button

#### 📷 Screenshots

image

### CODE REVIEW

File: src/app/components/register/register.component.ts [hunk 1]
```diff
@@ -57,9 +61,33 @@ export class RegisterComponent implements OnInit {
   }
 
   public onSubmit() {
-    this.toastService.error('Please fill the form');
     if (this.regForm.valid && this.agree) {
-      this.toastService.success('Success');
-    }
+      const dateOfBirth = new Date(this.regForm.value.date);
+      const isoDateOfBirth = dateOfBirth.toISOString();
+      console.log(
```
Comment (kubarenatioN) [inline src/app/components/register/register.component.ts]: Do we need this console.log?

File: src/app/services/auth.service.ts [hunk 2]
```diff
@@ -1,18 +1,76 @@
...
   }
+
+  public getToken(value: IRegUser) {
+    this.httpClient
+      .post<IToken>(`${this.url}/auth/registration`, JSON.stringify(value), {
+        headers: this.headers,
+      })
+      .subscribe(
+        (tokenObj) => {
+          localStorage.setItem('token', tokenObj.token);
```
Comment (kubarenatioN) [inline src/app/services/auth.service.ts:58]: Are we putting user password in localStorage? 😨

File: src/app/services/auth.service.ts [hunk 3]
```diff
@@ -1,18 +1,76 @@
...
+          localStorage.setItem('userName', value.firstName);
+          this.userName.next(localStorage.getItem('userName') as string);
+          this.userIsLogged.next(!!localStorage.getItem('token'));
+          this.dialog.closeAll();
+          this.toastService.success('Success');
+        },
+        (error) => {
+          if (error.message) {
+            console.log(JSON.stringify(error.error.message));
```
Comment (kubarenatioN) [inline src/app/services/auth.service.ts]: `error.message` is already a string, no need to stringify it

**Review by kubarenatioN — CHANGES_REQUESTED** [review]
Please attach screenshot to PR of final Register component form UI

File: src/app/services/auth.service.ts (hunk 2 above)
Comment (Zmicer-Yafremau) [reply src/app/services/auth.service.ts:58]: It's not a password, it's token

File: src/app/services/auth.service.ts (hunk 3 above)
Comment (Zmicer-Yafremau) [reply src/app/services/auth.service.ts]: done

File: src/app/components/register/register.component.ts (hunk 1 above)
Comment (Zmicer-Yafremau) [reply src/app/components/register/register.component.ts]: I've removed it

**Review by kubarenatioN — APPROVED** [review]

**Review by ShArPman13 — APPROVED** [review]

**Pull request merged**